### PR TITLE
refactor: useCreateArtworkAlert hook to reuse functionality

### PR DESCRIPTION
**Pull request by gkartalis**
This PR resolves [] 

### Description

Refactoring create artwork alert logic into a hook to be reused in other surfaces 

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- useCreateArtworkAlert hook to reuse functionality - gkartalis

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

**Review by MounirDhahri — DISMISSED** [review]
Thanks George!

File: src/app/utils/hooks/useCreateArtworkAlert.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+} from "app/Components/ArtworkFilter/SavedSearch/types"
+import { compact, isEmpty } from "lodash"
+import { useState } from "react"
+
+type Artwork =
+  | ArtworkScreenHeaderCreateAlert_artwork$data
+  | CreateArtworkAlertButtonsSection_artwork$data
+
+export const useCreateArtworkAlert = (artwork: Artwork) => {
+  const [isCreateAlertModalVisible, setIsCreateAlertModalVisible] = useState(false)
```
Comment (dimatretyak) [inline src/app/utils/hooks/useCreateArtworkAlert.ts:18]: **question**: what do you think about using `useFragmentContainer` here to query the necessary data inside a custom hook?

File: src/app/utils/hooks/useCreateArtworkAlert.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  const hasArtists = !isEmpty(artwork.artists) && artwork.artists!.length > 0
```
Comment (dimatretyak) [inline src/app/utils/hooks/useCreateArtworkAlert.ts:19]: **suggestion**: What do you think about this code?

```javascript
const artists = artwork.artists ?? []
const hasArtists = artists.length > 0
```

File: src/app/Scenes/Artwork/Components/ArtworkScreenHeaderCreateAlert.tsx [hunk 3]
```diff
@@ -28,63 +16,27 @@ const ArtworkScreenHeaderCreateAlert: React.FC<ArtworkScreenHeaderCreateAlertPro
...
   const hasArtists = !isEmpty(artwork.artists) && artwork.artists!.length > 0
 
+  const {
+    entity,
+    attributes,
+    aggregations,
+    showCreateArtworkAlertModal,
+    closeCreateArtworkAlertModal,
+    isCreateAlertModalVisible,
+  } = useCreateArtworkAlert(artwork)
```
Comment (dimatretyak) [inline src/app/Scenes/Artwork/Components/ArtworkScreenHeaderCreateAlert.tsx:28]: **question**: should we use `hasArtists` variable from `useCreateArtworkAlert` hook?

**Review by dimatretyak — APPROVED** [review]
LGTM 👍

File: src/app/utils/hooks/useCreateArtworkAlert.ts (hunk 1 above)
Comment (gkartalis) [reply src/app/utils/hooks/useCreateArtworkAlert.ts:18]: will def give it a try!

File: src/app/Scenes/Artwork/Components/ArtworkScreenHeaderCreateAlert.tsx (hunk 3 above)
Comment (gkartalis) [reply src/app/Scenes/Artwork/Components/ArtworkScreenHeaderCreateAlert.tsx:27]: I think I will keep it as is to have control in weird cases to let users create alert or not

**Review by dimatretyak — DISMISSED** [review]

File: src/app/Scenes/Artwork/Artwork.tests.tsx [hunk 4]
```diff
@@ -50,25 +50,20 @@ describe("Artwork", () => {
...
+        onLoad={onLoad}
+        pageableSlugs={[]}
+      />
+    </ModalStack>
   )
 
   beforeEach(() => {
-    require("app/system/relay/createEnvironment").reset()
-    environment = require("app/system/relay/createEnvironment").defaultEnvironment
+    environment = getMockRelayEnvironment()
```
Comment (gkartalis) [inline src/app/Scenes/Artwork/Artwork.tests.tsx:66]: that was the culprit 🤯

File: src/app/Scenes/Artwork/Artwork.tests.tsx [hunk 5]
```diff
@@ -50,25 +50,20 @@ describe("Artwork", () => {
...
-          isVisible={isVisible}
-          artworkID="ignored"
-          environment={environment}
-          tracking={{ trackEvent: jest.fn() } as any}
-          onLoad={onLoad}
-          pageableSlugs={[]}
-        />
-      </ModalStack>
-    </Suspense>
+    <ModalStack>
```
Comment (gkartalis) [inline src/app/Scenes/Artwork/Artwork.tests.tsx:53]: got rid of suspense because it wasnt used 🤷‍♂️

**Review by gkartalis — COMMENTED** [review]
while I was poking to fix the issue I also refactored some files to use useFragment vs container

**Review by damassi — APPROVED** [review]
Looks good 👍

**Pull request merged**